code/datums/helper_datums/teleport.dm

```diff
@@ -113,9 +113,76 @@
 		var/mob/living/L = teleatom
 		if(L.buckled)
 			C = L.buckled
+
 	if(force_teleport)
 		teleatom.forceMove(destturf)
 		playSpecials(destturf,effectout,soundout)
+		var/atom/impediment
+
+		if(destturf.density)
+			impediment = destturf
+
+		else
+			for(var/atom/A in destturf)
+				if(A.density)
+					impediment = A
+					break
+
+		if(impediment)
+			var/turf/newdest
+			var/boominess = 0
+			for(var/turf/T in range(1, destturf))
+				if(!T.density)
+					for(var/atom/A in T)
+						var/blocked = 0
+						if(A.density && A.opacity)
+							blocked = 1
+							break
+						if(!blocked)
+							newdest = T
+							break
+
+			if(istype(teleatom, /obj))
+				var/obj/O = teleatom
+				if(newdest)
+					O.forceMove(newdest)
+					O.ex_act(3)
+				else
+					O.crush_act()
+				boominess += O.w_class
+				if(O.density)
+					boominess += 5
+				if(O.opacity)
+					boominess += 10
+
+			if(istype(teleatom, /mob/living))
+				var/mob/living/L = teleatom
+				boominess += L.mob_size/4
+				if(istype(L, /mob/living/carbon/human))
+					var/mob/living/carbon/human/H = L
+					if(newdest)
+						var/obj/item/organ/external/E = pick(H.organs)
+						to_chat(H, "<span class='danger'>You partially phase into \the [impediment], causing your \the [E] to bloodily dematerialize!</span>")

```

Comment:
I dont think the second \the makes sense.
`causing your the right arm to bloodily dematerialize!`